### PR TITLE
CIV-14140 GA for LiP's - Authorise case worker to access GA application to allow to submit HWF events

### PR DESCRIPTION
**Pull request by manish14836**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-14140

### Change description ###

GA for LiP's - Authorise case worker to access GA application to allow to submit HWF events

### Testing branch and instruction ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/service/ParentCaseUpdateHelper.java [hunk 1]
```diff
@@ -236,24 +240,103 @@ protected String findGaCreator(CaseData civilCaseData, CaseData generalAppCaseDa
...
+
+            /**
+             * When main claim's 1 V 2 Same Legal Representative happens,
+             * Check if main claim "Respondent2SameLegalRespresentative" value is true,
+             * if so, ADD GA application has to master collection
+             *
+             * In addition to above, the condition : generalAppCaseData.getIsMultiParty().equals(NO)
+             * Add GA into mater collection if it's not multiparty scenario and GA initiated by Main claim Defendant 1v1
+             */
+        } else if ((Objects.nonNull(parentCaseData.getRespondent2SameLegalRepresentative())
```
Comment (manish14836) [inline src/main/java/uk/gov/hmcts/reform/civil/service/ParentCaseUpdateHelper.java:315]: Remove if

Comment (manish14836) [reply src/main/java/uk/gov/hmcts/reform/civil/service/ParentCaseUpdateHelper.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/genapplication/GeneralApplication.java [hunk 2]
```diff
@@ -122,7 +123,8 @@ public class GeneralApplication implements MappableObject {
                        @JsonProperty("generalAppHearingDate") GAHearingDateGAspec generalAppHearingDate,
                        @JsonProperty("isGaApplicantLip") YesOrNo isGaApplicantLip,
                        @JsonProperty("isGaRespondentOneLip") YesOrNo isGaRespondentOneLip,
-                       @JsonProperty("isGaRespondentTwoLip") YesOrNo isGaRespondentTwoLip) {
+                       @JsonProperty("isGaRespondentTwoLip") YesOrNo isGaRespondentTwoLip,
+                       @JsonProperty("generalAppHelpWithFees") HelpWithFees generalAppHelpWithFees) {
```
[the remark this answers is not in the document]
Comment (manish14836) [reply src/main/java/uk/gov/hmcts/reform/civil/model/genapplication/GeneralApplication.java:127]: Keep it in nonprod for now as access of it seems to be behind feature toggle

**Review by jarekPierchala — APPROVED** [review]

**Review by pliao-hmcts — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/ParentCaseUpdateHelper.java [hunk 3]
```diff
@@ -254,6 +254,83 @@ protected String findGaCreator(CaseData civilCaseData, CaseData generalAppCaseDa
...
+                                .generalApplicationType(generalApplicationsDetailsElement.getValue().getGeneralApplicationType())
+                                .generalAppSubmittedDateGAspec(generalApplicationsDetailsElement.getValue()
+                                                                   .getGeneralAppSubmittedDateGAspec())
+                                .caseLink(CaseLink.builder().caseReference(String.valueOf(
+                                    generalAppCaseData.getCcdCaseReference())).build()).build()));
+                }
+            });
+
+            /**
+             * When main claim's 1 V 2 Same Legal Representative happens,
```
Comment (neeta-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/service/ParentCaseUpdateHelper.java:304]: can be removed  as multiparty not supported

**Review by neeta-hmcts — DISMISSED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by sankhajuria — APPROVED** [review]

**Pull request merged**